proto/clustermanagerdata.proto

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Vitess Authors
+ Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreedto in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file contains vitess cluster manager related data structures.
+
+syntax = "proto3";
+option go_package = "vitess.io/vitess/go/vt/proto/clustermanagerdata";
+
+option java_package="io.vitess.proto";
+
+import "topodata.proto";
+
+package clustermanagerdata;
+
+// SetShardConfigRequest is the payload for SetShardConfigRequest
+message SetShardConfigRequest {
+  string cell = 1;
+  string keyspace = 2;
```

Comment:
Just string and not topodata.keyspace? 

Do you have to validate inside receiving the request that it's actually an existing cell/keyspace/shard?

Reply:
Yeah, I think topodata.Keyspace has way more information that we don't need in this context. Same for shard.